pkg/auth/native_passport.go

```diff
@@ -0,0 +1,70 @@
+package auth
+
+import (
+	"errors"
+
+	"github.com/urfave/cli/v2"
+)
+
+// NativePassport performs validation on credentials
+// and manipulates the credentials in the credentials store
+type NativePassport struct {
+	config    *ConfigFile
+	validator *AuthValidator
+}
+
```

Comment:
Most of time, we don't need a validator except login.

Try to reuse (and / or enhance) `notation-go/registry` for the registry client.

Reply:
Good call, I have refactored it.